sonic-net/sonic-buildimage
# Fix IPV6 forced-mgmt-route not work issue

Fix IPV6 forced-mgmt-route not work issue

#### Why I did it
IPV6 forced-mgmt-route not work

When add a IPV6 route, should use 'ip -6 rule add pref 32764 address' command, but currently in the template the '-6' parameter are missing, so the IPV6 route been add to IPV4 route table.

Also this PR depends on https://github.com/sonic-net/sonic-buildimage/pull/17281 , which will fix the IPV6 'default' route table missing in IPV6 route lookup issue. 

##### Work item tracking
- Microsoft ADO **(number only)**:24719238

#### How I did it
Add IPV6 route with 'ip -6' command


#### How to verify it
Pass all UT
Manually verify:

// setup forced mgmt routes
admin@vlab-01:~$ sonic-db-cli  CONFIG_DB HSET "MGMT_INTERFACE|eth0|fec0::ffff:afa:1/64" forced_mgmt_routes@ "1110::ffff:afa:1/64"
1
admin@vlab-01:~$ sudo config save -y
Running command: /usr/local/bin/sonic-cfggen -d --print-data > /etc/sonic/config_db.json
admin@vlab-01:~$ sudo config reload -y
Disabling container monitoring ...
Stopping SONiC target ...


// check interface config update correctly
admin@vlab-01:~$ cat /etc/network/interfaces

...

iface eth0 inet6 static
    address fec0::ffff:afa:1
    netmask 64
    network fec0::
    broadcast fec0::ffff:ffff:ffff:ffff
    ########## management network policy routing rules
    # management port up rules
    up ip -6 route add default via fec0::1 dev eth0 table default metric 201
    up ip -6 route add fec0::/64 dev eth0 table default
    up ip -6 rule add pref 32765 from fec0::ffff:afa:1/128 table default
    up ip **-6** rule add pref 32764 to 1110::ffff:afa:1/64 table default <== force mgmt route added correctly with 'ip -6'

// check route tables:
admin@vlab-01:~$ ip -6 rule list
1001:   from all lookup local
32764:  from all to 1110::ffff:afa:1/64 lookup default
32765:  from fec0::ffff:afa:1 lookup default <== force mgmt route been add to IPV6 table
32766:  from all lookup main
admin@vlab-01:~$


// traceroute command can find route
admin@vlab-01:~$ traceroute 1110::ffff:afa:1
traceroute to 1110::ffff:afa:1 (1110::ffff:afa:1), 30 hops max, 80 byte packets
 1  * * *
 2  * * *




#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)



- [ ] master-17299.420743-d31987daa

#### Description for the changelog
Fix IPV6 forced-mgmt-route not work issue

#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
